### PR TITLE
ghidra: reformat, 11.0.3 -> 11.1.1

### PR DESCRIPTION
**Pull request by jchv**
## Description of changes

Fairly straight-forward update of 11.0.3 to 11.1.1.

The patches still apply just fine, but they have some fuzz, which results in `.orig` files being added into the working directory, this breaks the build due to the IP check (which validates the licenses/copyrights allowed in Ghidra) because the file extension matters for this check. This is pretty undesirable so I add some `patchFlags` to prevent it, to reduce unnecessary churn in the patches.

Tested on `x86_64-linux` and `aarch64-darwin`. Tested loading and upgrading a project to 11.1.1 in code browser. Tested building and running with all Nixpkgs extensions and scripts, and ensuring that they show up as expected.

Upstream changelog: https://htmlpreview.github.io/?https://github.com/NationalSecurityAgency/ghidra/blob/Ghidra_11.1.1_build/Ghidra/Configurations/Public_Release/src/global/docs/ChangeHistory.html

Many improvements were made to the debugger, as well as accessibility in the UI, making it more usable for users who need to navigate with keyboard. The PDB analyzer is now split into multiple parts so that later portions of the analysis can benefit from work done in interim analyzers. Go 1.21 and 1.22 binaries are now supported, as well as AArch64/Apple Silicon Go binaries.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/security/ghidra/build.nix [hunk 1]
```diff
@@ -1,33 +1,35 @@
-{ stdenv
-, fetchFromGitHub
-, lib
-, callPackage
-, gradle_7
-, perl
-, makeBinaryWrapper
-, openjdk17
```
Comment (vringar) [inline pkgs/tools/security/ghidra/build.nix:8]: https://github.com/NationalSecurityAgency/ghidra/blob/5ab72bf4f254773b12c3c9566d14c42e5eb2495f/README.md?plain=1#L49-L50

We should probably also update to JDK 21 and Gradle 8.5+

File: pkgs/tools/security/ghidra/build.nix [hunk 2]
```diff
@@ -132,11 +137,23 @@ HERE
...
+    src
+    patches
+    postPatch
+    ;
+
+  # Don't create .orig files if the patch isn't an exact match.
+  patchFlags = [
+    "--no-backup-if-mismatch"
+    "-p1"
+  ];
```
Comment (vringar) [inline pkgs/tools/security/ghidra/build.nix:156]: Is it too much of a hassle to update the patches instead?

**Review by vringar — CHANGES_REQUESTED** [review]
Thank you so much for bumping this.

**Review by vringar — APPROVED** [review]

File: pkgs/tools/security/ghidra/build.nix (hunk 1 above)
Comment (myyra) [reply pkgs/tools/security/ghidra/build.nix:8]: Having just built 11.1.1 for myself, I can confirm it does.

**Review by liff — APPROVED** [review]
Built and tested, works fine.

**Pull request merged**